doc/introduction/interfaces/jax.rst

```diff
@@ -0,0 +1,137 @@
+
+
+.. note::
+
+    To use the JAX interface in PennyLane, you must first
+    install ``jax`` and ``jaxlib``. You can then import PennyLane and JAX as follows:
+
+    .. code::
+
+        import pennylane as qml
+        import jax
+        import jax.numpy as jnp
+
+
+Construction via the decorator
+------------------------------
+
+The :ref:`QNode decorator <intro_vcirc_decorator>` is the recommended way for creating
+a JAX-capable QNode in PennyLane. Simply specify the ``interface='jax'`` keyword argument:
+
+.. code-block:: python
+    
+    dev = qml.device('default.qubit.jax', wires=2)
+
+    @qml.qnode(dev, interface='jax')
+    def circuit1(phi, theta):
+        qml.RX(phi[0], wires=0)
+        qml.RY(phi[1], wires=1)
+        qml.CNOT(wires=[0, 1])
+        qml.PhaseShift(theta, wires=0)
+        return qml.expval(qml.PauliZ(0)), qml.expval(qml.Hadamard(1))
+
+The QNode ``circuit1()`` is now a JAX-capable QNode, accepting ``jax.DeviceArray`` objects
+as input, and returning ``jax.DeviceArray`` objects. It can now be used like any other JAX function:
+
+>>> phi = jnp.array([0.5, 0.1])
+>>> theta = jnp.array(0.2)
+>>> circuit1(phi, theta)
+DeviceArray([0.8776, 0.6880], dtype=float64)
+
+Quantum gradients using JAX
+---------------------------
+
+Since a JAX-interfacing QNode acts like any other JAX interfacing python function,
+the standard method used to calculate gradients with JAX can be used.
+
+For example:
+
+.. code-block:: python
+
+    dev = qml.device('default.qubit.jax', wires=2)
+
+    @qml.qnode(dev, interface='jax')
+    def circuit3(phi, theta):
+        qml.RX(phi[0], wires=0)
+        qml.RY(phi[1], wires=1)
+        qml.CNOT(wires=[0, 1])
+        qml.PhaseShift(theta, wires=0)
+        return qml.expval(qml.PauliZ(0))
+
+    phi = jnp.array([0.5, 0.1])
+    theta = jnp.array(0.2)
+    grads = jax.grad(circuit3, argnums=(0, 1))
+    phi_grad, theta_grad = grads(phi, theta)
+
+This has output:
+
+>>> phi_grad
+DeviceArray([-0.47942555,  0.        ], dtype=float32)
+>>> theta_grad
+DeviceArray(-3.4332792e-10, dtype=float32)
+
+
+.. _jax_optimize:
+
+Using jax.jit on QNodes
+-----------------------
+
+To fully utilize the power and speed of JAX, you'll need to just-in-time compile
+your functions. If you're only taking expectation values, you can simply add the
+decorator on your QNode directly.
+
+.. code-block:: python
+
+    dev = qml.device('default.qubit.jax', wires=2)
+    
+    @jax.jit  # QNode calls will now be jitted, and should run faster.
+    @qml.qnode(dev, interface='jax')
+    def circuit4(phi, theta):
+        qml.RX(phi[0], wires=0)
+        qml.RZ(phi[1], wires=1)
+        qml.CNOT(wires=[0, 1])
+        qml.RX(theta, wires=0)
+        return qml.expval(qml.PauliZ(0))
+
+
+However, if you want to do random sampling instead, you'll need to pass a ``jax.random.PRNGKey`` to the device construction. Your jitted function will also need to include the device construction.

```

Comment:
Worth inserting a link to a page that explains what the key does?

Reply:
I was confused about "include device construction". I think this needs to be reformulated entirely. Is what you want to say?

"If quantum nodes involve random sampling, jax requires a key called a ``PRNGKey'' to do XXX. The key can be passed to the device as a keyword argument: 

``qml.device(..., prng_key=key)`` 

However, now the jit decorator cannot be applied directly to the quantum function. This means that we have to wrap the qnode into another function which gets jitted with the decorator instead."

Also, this reads a bit like a hack - and maybe we should sell it as such and explain exactly why it requires such a detour with the wrapping function, and whether there are alternatives...

Reply:
> Also, this reads a bit like a hack

That's because it is a bit of a hack lol. But I think given how JAX's randomness is handled, it's likely the best we can do.

Reply:
Added some more context. let me know what you think!